cdap-app-fabric/src/test/java/io/cdap/cdap/scheduler/CoreSchedulerServiceTest.java

```diff
@@ -295,7 +296,12 @@ public void testRunScheduledJobs() throws Exception {
     int runs1 = getRuns(WORKFLOW_1, ProgramRunStatus.ALL);
     int runs2 = getRuns(WORKFLOW_2, ProgramRunStatus.ALL);
     disableSchedule(AppWithFrequentScheduledWorkflows.DATASET_PARTITION_SCHEDULE_1);
+
+    // ensure schedule 2 is disabled after schedule 1
+    Thread.sleep(BUFFER);
```

Comment:
I'm guessing this sleep is here because the method call on line 298 is actually asynchronous?
If so, can we poll on the status of schedule 1, instead of sleeping, since sleeping for 10ms might be prone to race condition, if running this test on a machine that is currently CPU-constrained.

Reply:
No, we poll on the status already within the function. I'm want to prevent the case where execution happens so fast that both are disabled within the same exact millisecond, which is probably very unlikely, but better safe than sorry